docs/development/secure-development-process.md

```diff
@@ -0,0 +1,196 @@
+  * One customer using too much -> External API manager (for example Red Hat 3scale API management)
+  * Oversized requests -> Enforcng request size limits
+* Prevent HTTP parameter polution with packages like
+  [hpp](https://www.npmjs.com/package/hpp)
+* Protect against Cross site scripting (XSS). For example by using a package like [xss](https://www.npmjs.com/package/xss).
+* Protect against cross site forgery requests
+  * Use Anti-CSRF tokens through packages like [csurf](https://www.npmjs.com/package/csurf)
+
+**Avoid leaking info through errors**
+
+Do not expose senstive or intermal information in errors returned to the end user. Instead return a unique ID that can be tied back to more specific error information that is written to internals logs.
+
+**Limit data sent back to front ends**
+
+Only send the required data to the front end. For example if a database query returns 4 fields but the front end will only display 3, make sure to remove the fourth field before sending the data on to the front end.
+
+**Discard sentive data after use**
+
+Do not retain senstive information beyond where it is needed. Overwrite fields/variables that contain sensitive data once it is not longer needed.
+
+**Validate input**
+
+Unvalidated input can result in:
+* command injection
+* sql injection
+* data corruption
+* denial of service
+
+Do not pass unsanitized user input to the Node.js [child_process](https://nodejs.org/api/child_process.html) functions as this can result in command injection where an attacker can run a common on the server on which Node.js is running. This is a good article that goes into more details: 
+[avoiding-arbitrary-code-execution-vulnerabilities-when-using-nodejs-child-process-apis](https://developer.ibm.com/articles/avoiding-arbitrary-code-execution-vulnerabilities-when-using-nodejs-child-process-apis/).
+
+Avoid using the [eval](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval) function. If you must use it, ensure that you have validate/sanitized all inputs that may be added to the string passed to eval.
+
+When passing user input to database clients always validate the inputs and use prepared statements instead of dynamically building requests (for example an SQL request string).
+
+Unvalidated json input if written directly to a database or accessed by your application can result in data corruption or denial of service when your application fails because it cannot access the fields it expects. Always validate JSON input with a package like [ajv](https://www.npmjs.com/package/ajv).
+
+**Use static analysis**
+
+As part of PR process process, incorporate static analysis. A number of the teams members have had success with `sonarcube` and `checkmarx` to complete these scans.
+
+### Limiting required execution privileges
+
+Design your application to run with the least priviledges required. Ensure your application can run as a non-root user, even when being deployed within containers.
```

Comment:
```
especially within a container, and add ref related discussion on building good containers.
Jim, need all tools, Anonthy this might help - https://loft.sh/blog/using-kubernetes-ephemeral-containers-for-troubleshooting/ 
```